Composer/packages/client/src/components/CreationFlow/index.tsx

```diff
@@ -0,0 +1,125 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+// TODO: Remove path module
+import Path from 'path';
+
+import React, { useEffect, useContext, useRef, Fragment } from 'react';
+import { RouteComponentProps, Router, navigate } from '@reach/router';
+
+import { CreationFlowStatus } from '../../constants';
+import { StoreContext } from '../../store';
+import Home from '../../pages/home';
+
+import { CreateOptions } from './CreateOptions';
+import { OpenProject } from './OpenProject';
+import DefineConversation from './DefineConversation';
+
+type CreationFlowProps = RouteComponentProps<{}>;
+
+const CreationFlow: React.FC<CreationFlowProps> = () => {
+  const { state, actions } = useContext(StoreContext);
+  const { creationFlowStatus } = state;
+  const {
+    fetchTemplates,
+    openBotProject,
+    createProject,
+    saveProjectAs,
+    saveTemplateId,
+    fetchStorages,
+    fetchFolderItemsByPath,
+    setCreationFlowStatus,
+  } = actions;
+  const { templateId, templateProjects, storages } = state;
+  const currentStorageIndex = useRef(0);
+  const storage = storages[currentStorageIndex.current];
+  const currentStorageId = storage ? storage.id : 'default';
+
+  useEffect(() => {
+    if (storages && storages.length) {
+      const storageId = storage.id;
+      const path = storage.path;
+      const formattedPath = Path.normalize(path);
+      fetchFolderItemsByPath(storageId, formattedPath);
+    }
+  }, [storages]);
+
+  useEffect(() => {
+    fetchStorages();
+    fetchTemplates();
+  }, []);
+
+  const updateCurrentPath = async (newPath, storageId) => {
+    if (!storageId) {
+      storageId = currentStorageId;
+    }
+    if (newPath) {
+      const formattedPath = Path.normalize(newPath);
+      await actions.updateCurrentPath(formattedPath, storageId);
+    }
+  };
+
+  const handleDismiss = () => {
+    setCreationFlowStatus(CreationFlowStatus.CLOSE);
+    navigate(`/home`);
+  };
+
+  const openBot = async botFolder => {
+    await openBotProject(botFolder);
+    handleDismiss();
+  };
+
+  const handleCreateNew = async formData => {
+    await createProject(templateId || '', formData.name, formData.description, formData.location, formData.schemaUrl);
+  };
+
+  const handleSaveAs = async formData => {
+    await saveProjectAs(state.projectId, formData.name, formData.description, formData.location);
+  };
+
+  const handleSubmit = formData => {
+    switch (creationFlowStatus) {
+      case CreationFlowStatus.NEW_FROM_SCRATCH:
+      case CreationFlowStatus.NEW_FROM_TEMPLATE:
+        handleCreateNew(formData);
+        break;
+      case CreationFlowStatus.SAVEAS:
+        handleSaveAs(formData);
+        break;
+
+      default:
+        handleCreateNew(formData);
+    }
+    handleDismiss();
+  };
```

Comment:
I feel this can be refactored. But I see quite a few usages of these in other components like toolbar, home, reducers etc. Some of those components are not completely unit tested for me to confidently remove creationStatus from the reducer and all usages of it everywhere. For ex if the creation status is being used for something else. It seems like a refactor that can hold after R9. what do u think?

Reply:
I'm not saying we need to refactor those completly away, but we don't need this switch statement in here. We can directly pass `handleCreateNew` or `handleSaveAs` to the components that require them.